### PR TITLE
Main

### PR DESCRIPTION
**Pull request by maciejbentkowski**
Everything works but i know some ways of manipulating data in databases needs refactor. Im looking for learning materials how to organize my code

### CODE REVIEW

File: spec/system/order_system_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+RSpec.describe 'Order', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  describe 'Points' do
+    before do
+      visit '/employees/sign_in'
+      fill_in 'Email', with: employee.email
```
Comment (rniedziolka) [inline spec/system/order_system_spec.rb:11]: I would suggest using capybara login helper.

File: spec/system/order_system_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    it 'Remove points after buy a reward' do
+      expect(employee.points).to eq(20.0)
+      employee_old_points = employee.points
+      visit '/orders/new'
+      select employee.email, from: 'Employee'
+      select reward.title, from: 'Reward'
+      click_button 'commit'
+      expect(page).to have_content 'You bought reward succesfully'
+      employee.reload
+      expect(employee.points).to eq(employee_old_points - reward.price)
```
Comment (rniedziolka) [inline spec/system/order_system_spec.rb:28]: My suggestion is to test the option when you have no points to buy a reward, which will then happen.

File: app/views/rewards/index.html.erb [hunk 3]
```diff
@@ -4,14 +4,23 @@
...
-</table>
\ No newline at end of file
+</table>
+
+<h2>Your rewards</h2>
+
+<% @orders.each do |order| %>
+  <td><%= order.reward.title %></td>
+  <td><%= order.employee.email %></td></br>
+<%end %>
```
Comment (rniedziolka) [inline app/views/rewards/index.html.erb:26]: Missing end line.

**Comment by rniedziolka** [comment]
Hi Maciej,
Nice to e-meet you! 😄 
Nice one!
In my opinion, I miss a place where it would display placed orders, I know it wasn't in AC, but it just seemed intuitive to me.
I think you could use a nicer name for PRs and Branches - take a look here: https://dev.to/couchcamote/git-branching-name-convention-cch
These are just my suggestions please wait for the comments of professionals.

File: spec/system/order_system_spec.rb (hunk 2 above)
Comment (ciejj) [reply spec/system/order_system_spec.rb:28]: yeah, there should be 2 examples: one for successful purchase (when an employee has enough points), and another one for buying without enough points

File: spec/system/order_system_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe 'Order', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  describe 'Points' do
```
Comment (ciejj) [inline spec/system/order_system_spec.rb:8]: I suggest something a bit more descriptive
```suggestion
  describe 'Ordering Rewards' do
```

**Review by ciejj — APPROVED** [review]
Good job.
Consider getting rid of that callbacks/ custom validations - they are actually not used that often, and make code harder to read :)

**Comment by maciejbentkowski** [comment]
@rniedziolka @ciejj Thanks for Your suggestions. I would try to implement them with next PR

**Pull request merged**